decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/handlers/Bcl2FastqHandler.java

```diff
@@ -160,58 +154,6 @@ public WorkflowRunV2 getWorkflowRun(Metadata metadata, Bcl2FastqData data, boole
         return wr;
     }
 
-    public static String calculateBasesMask(List<String> barcodes) throws DataMismatchException {
-
-        if (barcodes == null || barcodes.isEmpty()) {
-            return null;
-        }
-
-        Set<Boolean> isNoIndexSet = new HashSet<>();
-        for (String barcode : barcodes) {
-            isNoIndexSet.add(barcode == null || barcode.isEmpty() || "NoIndex".equals(barcode));
-        }
-
-        Set<Boolean> isDualBarcodeSet = new HashSet<>();
-        for (String barcode : barcodes) {
-            isDualBarcodeSet.add(barcode != null && barcode.contains("-"));
-        }
-
-        Set<Integer> barcodeLengths = new HashSet<>();
-        for (String barcode : barcodes) {
-            if (barcode == null || barcode.isEmpty() || "NoIndex".equals(barcode)) {
-                barcodeLengths.add(0);
-            } else {
-                barcodeLengths.add(barcode.length());
-            }
-        }
-
-        if (isNoIndexSet.size() != 1) {
-            throw new DataMismatchException("Combination of NoIndex and barcoded samples");
-        }
-
-        if (isDualBarcodeSet.size() != 1) {
-            throw new DataMismatchException("Combination of dual-barcode and non-dual-barcode");
-        }
-
-        if (barcodeLengths.size() != 1) {
-            throw new DataMismatchException("Combination of barcodes with different lengths");
-        }
-
-        boolean isNoIndex = Iterables.getOnlyElement(isNoIndexSet);
-        boolean isDualBarcode = Iterables.getOnlyElement(isDualBarcodeSet);
-        int barcodeLength = Iterables.getOnlyElement(barcodeLengths);
-
-        if (isNoIndex) {
-            return null;
-        } else if (isDualBarcode) {
-            return "y*,I*,I*,y*";
-        } else if (barcodeLength > 0) {
-            return "y*,I" + barcodeLength + "n*,y*";
-        } else {
-            throw new DataMismatchException("Unable to calculate bases mask");
-        }
-    }
-
     private String generateLinkingString(IusWithProvenance<ProvenanceWithProvider<LaneProvenance>> lane, List<IusWithProvenance<ProvenanceWithProvider<SampleProvenance>>> sps, boolean enableDemultiplexSingleSample) throws DataMismatchException {
         List<String> sampleStringEntries = new ArrayList<>();
         Collections.sort(sps, new Comparator<IusWithProvenance<ProvenanceWithProvider<SampleProvenance>>>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)